src/main/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGenerator.java

```diff
@@ -0,0 +1,355 @@
+ * so they can use them in their configuration gui.
+ * @author Pavel Bludov
+ */
+public final class JavadocPropertiesGenerator {
+
+    /**
+     * The command line option to specify the output file.
+     */
+    private static final String OPTION_DEST_FILE = "destfile";
+
+    /**
+     * The width of the CLI help option.
+     */
+    private static final int HELP_WIDTH = 100;
+
+    /**
+     * This regexp is used to extract the first sentence from the text.
+     * The end of the sentence is determined by the symbol "period", "exclamation mark" or
+     * "question mark", followed by a space or the end of the text.
+     */
+    private static final Pattern END_OF_SENTENCE_PATTERN = Pattern.compile("(.*?[.?!])(\\s|$)");
+
+    /**
+     * Don't create instance of this class, use the {@link #main(String[])} method instead.
+     */
+    private JavadocPropertiesGenerator() {
+    }
+
+    /**
+     * TokenTypes.properties generator entry point.
+     * @param args the command line arguments
+     * @throws CheckstyleException if parser or lexer failed or if there is an IO problem
+     * @throws ParseException if the command line can not be passed
+     **/
+    public static void main(String... args)
+            throws CheckstyleException, ParseException {
+        final CommandLine commandLine = parseCli(args);
+        if (commandLine.getArgList().size() == 1) {
+            final String inputFile = commandLine.getArgList().get(0);
+            final String outputFile = commandLine.getOptionValue(OPTION_DEST_FILE);
+            writePropertiesFile(inputFile, outputFile);
+        }
+        else {
+            printUsage();
+        }
+    }
+
+    /**
+     * Creates the .properties file from a .java file.
+     * @param inputFile .java file
+     * @param outputFile .properties file
+     * @throws CheckstyleException if a javadoc comment can not be parsed
+     * @noinspection IOResourceOpenedButNotSafelyClosed
+     */
+    private static void writePropertiesFile(String inputFile, String outputFile)
+            throws CheckstyleException {
+        FileOutputStream fos = null;
+        PrintWriter writer = null;
+        try {
+            fos = new FileOutputStream(outputFile);
+            final Writer osw = new OutputStreamWriter(fos, StandardCharsets.UTF_8);
+            writer = new PrintWriter(osw, false);
+
+            final DetailAST top = JavaParser.parseFile(new File(inputFile),
+                JavaParser.Options.WITH_COMMENTS);
+            final DetailAST objBlock = getClassBody(top);
+            if (objBlock != null) {
+                iteratePublicStaticIntFields(objBlock, writer::println);
+            }
+        }
+        catch (IOException ex) {
+            throw new CheckstyleException("Failed to write javadoc properties of '" + inputFile
+                + "' to '" + outputFile + "'", ex);
+        }
+        finally {
+            CommonUtils.close(writer);
+            CommonUtils.close(fos);
```

Comment:
I don't like this. If `new FileOutputStream(outputFile)` throws an exception, there is nothing to close in fos.
If not, the writer will be created and after `CommonUtils.close(writer)` there will be nothing to close either.

Here's another version. Still not perfect, though.